presto-spi/src/main/java/com/facebook/presto/spi/statistics/ColumnStatistics.java

```diff
@@ -70,31 +53,35 @@ public static Builder builder()
 
     public static final class Builder
     {
-        private Estimate dataSize = unknownValue();
-        private Estimate nullsFraction = unknownValue();
-        private Estimate distinctValuesCount = unknownValue();
+        private List<RangeColumnStatistics> rangeColumnStatistics = new ArrayList<>();
 
-        public Builder setDataSize(Estimate dataSize)
+        public Builder addRange(Consumer<RangeColumnStatistics.Builder> rangeBuilderConsumer)
         {
-            this.dataSize = requireNonNull(dataSize, "dataSize can not be null");
+            RangeColumnStatistics.Builder rangeBuilder = RangeColumnStatistics.builder();
+            rangeBuilderConsumer.accept(rangeBuilder);
+            addRange(rangeBuilder.build());
             return this;
         }
 
-        public Builder setNullsFraction(Estimate nullsFraction)
+        public Builder addRange(Object lowValue, Object highValue, Consumer<RangeColumnStatistics.Builder> rangeBuilderConsumer)
```

Comment:
I assume that this variant of addRange is here for purposes of providing more options of communication for SPI users, as this is not used at this point.